internal/repository/sql/resource_repository.go

```diff
@@ -163,30 +181,49 @@ func applyQuery(db *gorm.DB, query repository.Query) *gorm.DB {
 		query.Limit = repository.DefaultPaginationLimit
 	}
 
-	return handlePagination(query.Paginator, db).Limit(query.Limit)
+	return handlePagination(query.Paginator, db).Limit(query.Limit), nil
 }
 
 // handleCompositeKey applies the composite key to the query.
-func handleCompositeKey(db *gorm.DB, compositeKey repository.CompositeKey) *gorm.DB {
+func handleCompositeKey(db *gorm.DB, compositeKey repository.CompositeKey) (*gorm.DB, error) {
 	tx := db.Session(&gorm.Session{NewDB: true})
 
 	for field, value := range compositeKey {
-		switch value {
-		case repository.NotEmpty:
-			tx = tx.Where(field+" IS NOT NULL").Where(field+" != ?", "")
-		case repository.Empty:
-			tx = tx.Where(field+" IS NULL OR "+field+" = ?", "")
-		default:
-			v := reflect.ValueOf(value)
-			if v.Kind() == reflect.Slice || v.Kind() == reflect.Array {
-				tx = tx.Where(field+" IN ?", value)
-			} else {
-				tx = tx.Where(field+" = ?", value)
-			}
+		var err error
+		tx, err = handleQueryField(value, tx, field)
+		if err != nil {
+			return nil, err
 		}
 	}
 
-	return tx
+	return tx, nil
+}
+
+// handleQueryField applies the query field to the query.
+func handleQueryField(value any, tx *gorm.DB, field repository.QueryField) (*gorm.DB, error) {
```

Comment:
```suggestion
func buildQuery(tx *gorm.DB,  field repository.QueryField, value any) *gorm.DB {
```
@fabenan-f and me think ordering of the func parameters would be nice.